### PR TITLE
[REF] Clean up on $ids['contribution']

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
[REF] Clean up on $ids['contribution']

Before
----------------------------------------
- $ids['contribution'] is now just an internal parameter since pass-by-ref is gone but it's still being stored in the incoming array $ids
- 2 if clauses are both dependent on there being a contribution id - but it's hard to tell

After
----------------------------------------
var in use is $contributionID - the 2 ifs are wrapped in if ($contributionID) and it's moved out of the individual conditions, loading of line items to support the ifs is also moved in there

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Member/BAO/Membership.php [hunk 1]
```diff
@@ -352,41 +352,45 @@ public static function create(&$params, $ids = []) {
...
-            $params['line_item'][$priceSetId][$lineIndex]['entity_id'] = $membership->id;
-            $params['line_item'][$priceSetId][$lineIndex]['entity_table'] = 'civicrm_membership';
-          }
-          elseif (!$lineMembershipType && !empty($params['contribution'])) {
-            $params['line_item'][$priceSetId][$lineIndex]['entity_id'] = $params['contribution']->id;
-            $params['line_item'][$priceSetId][$lineIndex]['entity_table'] = 'civicrm_contribution';
+    // If the membership has no associated contribution then we ensure
+    // the line items are 'correct' here. This is a lazy legacy
+    // hack whereby they are deleted and recreated
+    if (empty($contributionID)) {
```
Comment (eileenmcnaughton) [inline CRM/Member/BAO/Membership.php:358]: Read this with w=1 due to a big IF change

https://github.com/civicrm/civicrm-core/pull/19207/files?w=1

File: CRM/Member/BAO/Membership.php [hunk 2]
```diff
@@ -352,41 +352,45 @@ public static function create(&$params, $ids = []) {
...
+      if (!empty($params['lineItems'])) {
```
Comment (seamuslee001) [inline CRM/Member/BAO/Membership.php:359]: @eileenmcnaughton looking at these 3 lines specifically, There does seem to be a change in that before this change, this specific if would be run whether there was a ContributionID found or not. Are you sure this won't be needed for the `createRelatedMemberships` function at all?

Comment (eileenmcnaughton) [reply CRM/Member/BAO/Membership.php:359]: @seamuslee001 that function actively unsets them

https://github.com/civicrm/civicrm-core/blob/bcbc2c645effde0fb685f294294cd9d216e412ca/CRM/Member/BAO/Membership.php#L1460-L1463

Comment (seamuslee001) [reply CRM/Member/BAO/Membership.php:359]: ok

**Comment by seamuslee001** [comment]
Change made sense to me, just had the one question which Eileen has now responded to merging

**Pull request merged**